src/checkers/inference/InferenceTreeAnnotator.java

```diff
@@ -182,17 +183,30 @@ public Void visitIdentifier(IdentifierTree node, AnnotatedTypeMirror identifierT
 
                 } else if (parentNode.getKind() == Kind.NEW_CLASS
                         && ((NewClassTree) parentNode).getIdentifier() == node) {
-                    // This can happen at two locations in a NewClassTree:
-                    // (1) The type identifier of the NewClassTree, as `A` of `new A() {}`,
-                    // (2) The type identifier on the anonymous class's extends/implements clause.
-                    // Note that the identifiers trees described in the two cases above are identical
-                    // for one NewClassTree, i.e. they share the same slot
-                    // TODO: A NewClassTree should be handled in visitNewClass method exclusively 
+                    // This can happen in two cases related to NewClassTrees:
+                    // (1) The type identifier of non-anonymous class instantiations, without explict
+                    //     annotations, such as `A` of `new A()`;
+                    // (2) The type identifier of anonymous class instantiations, with or without
+                    //     explicit annotations.
+                    // TODO: A NewClassTree should be handled in visitNewClass method exclusively
                     // without messing around with the IdentifierTree or ClassTree, see issue:
                     // https://github.com/opprop/checker-framework-inference/issues/332
-                    variableAnnotator.visit(identifierType, node);
 
+                    NewClassTree newClassTree = (NewClassTree) parentNode;
+                    if (newClassTree.getClassBody() != null) {
+                        // For case 2, get the explicit annotation if any exists so that no variable slot
```

Comment:
Can you clarify what the "no variable slot is created" means relative to the code in `fromNewClass` above that creates a slot? Is the `type` in `fromNewClass` the `identifierType` here? So this passes the explicit annotation so that it will later be turned into a slot?